src/index.ts

```diff
@@ -218,8 +131,20 @@ export type FileSystemShortHands = {
  * while `assemble` does not. Use the latter in case you want to bypass the indexedDB check,
  * which might not be needed, or available, in certain environments or using certain components.
  */
-export async function program(settings: Partial<Components> & Configuration): Promise<Program> {
-  if (!settings) throw new Error("Expected a settings object of the type `Partial<Components> & Configuration` as the first parameter")
+export async function program<Annex extends AnnexParentType>(
+  settings: Partial<Components<Annex>> & { account: Account.Implementation<Annex> } & Configuration
+): Promise<Program<Annex>>
+export async function program(
+  settings: Partial<Omit<Components<FissionAccountsAnnex>, "account">> & Configuration
+): Promise<Program<FissionAccountsAnnex>>
+export async function program(
+  settings: Partial<Components<any>> & Configuration
+): Promise<Program<any>> {
```

Comment:
Can we get rid of these `any`s? Is it possible to use the `Annex extends Account.AnnexParentType` used in the `Program` type?

Reply:
I haven't found a way to get this working without `any` sadly. It should be fine though because this is the implementor type, Typescript will be mainly looking at types preceding this type (last type is a special case).

Reply:
I'll add though that this use of `any` is not that big of deal because it is localized and won't be consumed elsewhere.

Reply:
> Not entirely sure this would work, but maybe.

Yeah tried that, doesn't seem to work.